sdk/messaging/eventgrid/aznamespaces/client_custom.go

```diff
@@ -100,3 +111,140 @@ func (client *Client) ReleaseCloudEvents(ctx context.Context, topicName string,
+	if topicName == "" {
+		return nil, errors.New("parameter topicName cannot be empty")
+	}
+	urlPath = strings.ReplaceAll(urlPath, "{topicName}", url.PathEscape(topicName))
+	req, err := runtime.NewRequest(ctx, http.MethodPost, runtime.JoinPaths(client.endpoint, urlPath))
+	if err != nil {
+		return nil, err
+	}
+	reqQP := req.Raw().URL.Query()
+	reqQP.Set("api-version", "2023-10-01-preview")
+	req.Raw().URL.RawQuery = reqQP.Encode()
+	req.Raw().Header["Accept"] = []string{"application/json"}
+
+	req.Raw().Header.Set("ce-id", event.ID)
+	req.Raw().Header.Set("ce-specversion", event.SpecVersion)
+	req.Raw().Header.Set("ce-time", event.Time.UTC().Format(time.RFC3339))
+	req.Raw().Header.Set("ce-source", event.Source)
+
+	if event.Subject != nil {
+		req.Raw().Header.Set("ce-subject", *event.Subject)
+	}
+
+	req.Raw().Header.Set("ce-type", event.Type)
+
+	if event.DataSchema != nil {
+		req.Raw().Header.Set("ce-dataschema", *event.DataSchema)
+	}
+
+	contentType := ""
+
+	if event.DataContentType != nil {
+		contentType = *event.DataContentType
+	}
+
+	asBytes, ok := event.Data.([]byte)
+
+	if !ok {
+		return nil, fmt.Errorf("CloudEvent.Data must be of type []byte, was type %T", event.Data)
+	}
+
+	bodyStream := streaming.NopCloser(bytes.NewReader(asBytes))
+
+	// encode custom headers
+	for k, v := range event.Extensions {
+		// https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#type-system
+		headerName := "ce-" + k
+
+		str, err := stringize(v)
+
+		if err != nil {
+			return nil, err
+		}
+
+		req.Raw().Header.Set(headerName, str)
+	}
+
+	if err := req.SetBody(bodyStream, contentType); err != nil {
+		return nil, err
+	}
+
+	return req, nil
+}
+
+// stringize converts an arbitrary value into an equivalent string, complying with the types
+// in the CloudEvents spec: https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#type-system
+func stringize(tmpV any) (string, error) {
+	switch v := tmpV.(type) {
+	case bool:
+		return strconv.FormatBool(v), nil
+
+	// signed integer
+	case int8:
+		return strconv.FormatInt(int64(v), 10), nil
+	case int16:
+		return strconv.FormatInt(int64(v), 10), nil
+	case int:
+		return strconv.FormatInt(int64(v), 10), nil
+	case int32:
+		return strconv.FormatInt(int64(v), 10), nil
+	case int64:
+		return strconv.FormatInt(int64(v), 10), nil
+
+	// unsigned integers
+	case uint8: // (also handles `byte`)
+		return strconv.FormatUint(uint64(v), 10), nil
+	case uint16:
+		return strconv.FormatUint(uint64(v), 10), nil
+	case uint:
+		return strconv.FormatUint(uint64(v), 10), nil
+	case uint32:
+		return strconv.FormatUint(uint64(v), 10), nil
+	case uint64:
+		return strconv.FormatUint(uint64(v), 10), nil
+
+	case []byte:
+		return base64.StdEncoding.EncodeToString(v), nil
+	case *url.URL:
+		return v.String(), nil
+	case string:
+		return v, nil
+
+	case time.Time:
+		return v.Format(time.RFC3339), nil
+	case *time.Time:
+		return v.Format(time.RFC3339), nil
+
+	case fmt.Stringer:
```

Comment:
I think this will check if the type of `v` is `fmt.Stringer`, not that `v` implements it? If so, you should pull this out of the `switch` and check if `v` implements `fmt.Stringer` first. It would also allow you to remove the case for `*url.URL`.

Reply:
It is doing an "implements" check (just verified it since I can remove the url.URL case and it properly works).

Interestingly I had a hard time finding something saying that in the Go docs but there is this bit where they write the equivalent check in if/else statements that shows that 'case <interface>' does type assertions, which is what we were hoping for:

https://go.dev/ref/spec#:~:text=%7D%20else%20if%20i%2C%20isFloat64%20%3A%3D%20v.(float64)%3B%20isFloat64%20%7B